lib/platform-independent/state-machine.cpp

```diff
@@ -143,6 +143,11 @@ uint16_t get_original_time(state_machine_t *sm)
     return sm->timer.original_time;
 }
 
+uint16_t get_current_time(state_machine_t *sm)
+{
+    return sm->timer.current_time;
```

Comment:
Here you should create a getter function instead of reading the internal field directly (law of Demeter).

```suggestion
    return timer_get_current_time(&sm->timer);
```

This makes the state machine independent of any refactorings that we may do in the timer class - let's say we find a bug in the timer, and have to change it from counting up instead of down, or if we need to change the internal resolution to fix rounding, or something like that.

Reply:
Resolved in 769a871